src/ZkFold/Symbolic/Data/Combinators.hs

```diff
@@ -13,6 +15,12 @@ import qualified Prelude                         as Haskell
 
 import           ZkFold.Base.Algebra.Basic.Class
 
+-- | A class for isomorphic types.
+-- The @Iso b a@ context ensures that transformations in both directions are defined
+--
+class Iso b a => Iso a b where
```

Comment:
@vks4git Shouldn't there also be `into` method since you need a reverse morphism?

Reply:
It is implied by the superclass constraint

Reply:
Nope, `Iso b a` is listed as context, it will have the inverse mapping. You won't be able to define `Iso a b` without defining `Iso b a`, GHC won't compile such code

ghci> import ZkFold.Symbolic.Data.Combinators 
ghci> {instance Iso Bool (Maybe ()) where; from True = Just (); from False = Nothing}

<interactive>:2:11: error: [GHC-39999]
    • No instance for ‘Iso (Maybe ()) Bool’
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘Iso Bool (Maybe ())’